File: ch05_thermalExpansion/matlab/thexp_study_plot.m
clear
close all

fname = '../data/thexp_study.csv';
data = csvread(fname);

LW = 2;
kref = 1.1829318272893925;

temperature = data(:,1);
k = data(:,2);

figure
plt = plot(temperature,k,'-o','LineWidth',LW);
set(plt,'MarkerFaceColor',get(plt,'Color'));
title('Thermal Expansion Reactivity Study')
xlabel('Thermal Expansion Temperature [K]')
ylabel('k_{eff}')
set(gca,'FontName','Times New Roman')
print(gcf,'../figs/thexp_study.eps','-depsc2');
close(gcf)

rho = (k - kref)./(k * kref) * 1e5;
figure
plt = plot(temperature,rho,'-o','LineWidth',LW);
set(plt,'MarkerFaceColor',get(plt,'Color'));
title('Thermal Expansion Reactivity Study')
xlabel('Thermal Expansion Temperature [K]')
ylabel('\Delta \rho [pcm]')
set(gca,'FontName','Times New Roman')
print(gcf,'../figs/thexp_study_reactivity.eps','-depsc2');
close(gcf)

File: ch05_thermalExpansion/python/thexp_study_plot.py
import numpy as np
import matplotlib.pyplot as plt

LW = 2
FN = "Times New Roman"
FS = 12
plt.rc("lines", lw=LW)
plt.rc("mathtext", fontset="stix")
plt.rc("font", family=FN, size=FS)

kref = 1.1829318272893925

fname = "../data/thexp_study.csv"
data = np.loadtxt(fname, delimiter=",")
temperature = data[:, 0]
k = data[:, 1]

plt.figure()
plt.plot(temperature, k, "-o")
plt.title("Thermal Expansion Reactivity Study")
plt.xlabel("Thermal Expansion Temperature [K]")
plt.ylabel("$k_{eff}$")
plt.tight_layout()
plt.savefig("../figs/thexp_study.pdf")

rho = (k - kref) / (k * kref) * 1e5

plt.figure()
plt.plot(temperature, rho, "-o")
plt.title("Thermal Expansion Reactivity Study")
plt.xlabel("Thermal Expansion Temperature [K]")
plt.ylabel("$\Delta \\rho$ [pcm]")
plt.tight_layout()
plt.savefig("../figs/thexp_study_reactivity.pdf")
